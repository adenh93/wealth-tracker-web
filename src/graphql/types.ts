export type Maybe<T> = T | null;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type CryptoCurrency = {
  __typename?: 'CryptoCurrency';
  id: Scalars['Int'];
  name: Scalars['String'];
  rank: Scalars['Int'];
  symbol: Scalars['String'];
};

export type CryptoCurrencyHolding = {
  __typename?: 'CryptoCurrencyHolding';
  cryptoCurrency?: Maybe<CryptoCurrency>;
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
  id: Scalars['Int'];
  netHoldingsValue: Scalars['Float'];
  percentChange24h: Scalars['Float'];
  price: Scalars['Float'];
};

export type CryptoCurrencyHoldingsAddInput = {
  cryptoCurrencyId: Scalars['Int'];
  holdings: Scalars['Float'];
};

export type CryptoCurrencyHoldingsUpdateInput = {
  holdings: Scalars['Float'];
  id: Scalars['Int'];
};

export type Mutation = {
  __typename?: 'Mutation';
  cryptoCurrencyHoldingsAdd?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsDelete?: Maybe<CryptoCurrencyHolding>;
  cryptoCurrencyHoldingsUpdate?: Maybe<CryptoCurrencyHolding>;
};


export type MutationCryptoCurrencyHoldingsAddArgs = {
  input: CryptoCurrencyHoldingsAddInput;
};


export type MutationCryptoCurrencyHoldingsDeleteArgs = {
  id: Scalars['Int'];
};


export type MutationCryptoCurrencyHoldingsUpdateArgs = {
  input: CryptoCurrencyHoldingsUpdateInput;
};

export type Query = {
  __typename?: 'Query';
  cryptoCurrencies?: Maybe<Array<CryptoCurrency>>;
  cryptoCurrencyHoldings?: Maybe<Array<CryptoCurrencyHolding>>;
};


export type QueryCryptoCurrenciesArgs = {
  query?: Maybe<Scalars['String']>;
};
