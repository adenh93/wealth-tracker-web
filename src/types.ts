export interface Asset {
  id: string
  logoSrc: string
  name: string
  ticker: string
  price: number
  gain24Hr: number
  holdings: number
}
