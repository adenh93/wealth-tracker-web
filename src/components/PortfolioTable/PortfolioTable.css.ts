import { styled } from '@mui/material/styles'
import { TableHead } from '@mui/material'

export const StyledTableHead = styled(TableHead)(({ theme }) => ({
  backgroundColor: theme.palette.primary.main,
}))
