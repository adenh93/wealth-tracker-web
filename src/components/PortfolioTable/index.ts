export { default } from './PortfolioTable'
export type { PortfolioTableProps } from './PortfolioTable'