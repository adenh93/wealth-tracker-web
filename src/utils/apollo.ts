import { ApolloError } from '@apollo/client'

export const extractInputErrorField = <FormType>(
  error: ApolloError
): keyof FormType | null => {
  if (error.graphQLErrors.length)
    return error.graphQLErrors[0].extensions?.field as keyof FormType
  return null
}
