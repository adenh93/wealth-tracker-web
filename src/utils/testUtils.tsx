import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
