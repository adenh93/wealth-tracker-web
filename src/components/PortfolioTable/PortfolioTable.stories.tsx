import { Story } from '@storybook/react'
import { Asset } from '../../types'
import PortfolioTable, { PortfolioTableProps } from './PortfolioTable'

const config = {
  title: 'Components/PortfolioTable',
  component: PortfolioTable,
}

const Template: Story<PortfolioTableProps> = (args) => (
  <PortfolioTable {...args} />
)

export const Primary = Template.bind({})

const assets: Asset[] = [
  {
    logoSrc: 'http://simpleicon.com/wp-content/uploads/apple.png',
    name: 'Apple Inc.',
    ticker: 'AAPL',
    price: 148.98,
    gain24Hr: 0.012,
    holdings: 1350,
  },
  {
    logoSrc:
      'https://upload.wikimedia.org/wikipedia/commons/4/4e/BitConnect.png',
    name: 'Bitconnect',
    ticker: 'BCC',
    price: 0.003021,
    gain24Hr: -0.9984,
    holdings: 2450000.15,
  },
]

Primary.args = {
  assets,
}

export default config
