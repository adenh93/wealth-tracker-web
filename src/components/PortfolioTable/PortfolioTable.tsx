import { FC } from 'react'
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
  Paper,
} from '@mui/material'
import { OwnedAsset } from '../../types'
import PortfolioAsset from '../PortfolioAsset'
import TableHeadText from '../typography/TableHeadText'
import { StyledTableHead } from './PortfolioTable.css'

export interface PortfolioTableProps {
  assets: OwnedAsset[]
}

const PortfolioTable: FC<PortfolioTableProps> = ({ assets }) => (
  <TableContainer component={Paper}>
    <Table>
      <StyledTableHead>
        <TableRow>
          <TableCell align="left">
            <TableHeadText>Asset</TableHeadText>
          </TableCell>
          <TableCell align="left">
            <TableHeadText>Price</TableHeadText>
          </TableCell>
          <TableCell align="left">
            <TableHeadText>Holdings</TableHeadText>
          </TableCell>
          <TableCell />
        </TableRow>
      </StyledTableHead>
      <TableBody>
        {assets.map((asset: OwnedAsset) => (
          <PortfolioAsset key={asset.id} asset={asset} />
        ))}
      </TableBody>
    </Table>
  </TableContainer>
)

export default PortfolioTable
