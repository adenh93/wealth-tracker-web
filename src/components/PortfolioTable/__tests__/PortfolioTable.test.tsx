import PortfolioTable from '..'
import { Asset } from '../../../types'
import { render } from '../../../utils/testUtils'

describe('PortfolioTable component tests', () => {
  const mockAssets: Asset[] = [
    {
      id: 'test-id-1',
      logoSrc: 'https://localhost:3000/test.png',
      name: 'Apple Inc.',
      ticker: 'AAPL',
      price: 148.98,
      gain24Hr: 0.012,
      holdings: 1350,
    },
    {
      id: 'test-id-2',
      logoSrc: 'https://localhost:3000/test.png',
      name: 'Bitconnect',
      ticker: 'BCC',
      price: 0.003021,
      gain24Hr: -0.9984,
      holdings: 2450000.15,
    },
  ]

  test('matches snapshot', () => {
    const { asFragment } = render(<PortfolioTable assets={mockAssets} />)

    expect(asFragment).toMatchSnapshot()
  })

  test('it renders correctly', () => {
    const { container, getByText } = render(
      <PortfolioTable assets={mockAssets} />
    )

    const rows = container.querySelectorAll('tbody tr')

    expect(rows).toHaveLength(2)
    expect(getByText('Apple Inc.')).not.toBeNull()
    expect(getByText('Bitconnect')).not.toBeNull()
  })
})
