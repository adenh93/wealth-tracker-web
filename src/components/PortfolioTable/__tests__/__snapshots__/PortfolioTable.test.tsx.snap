// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioTable component tests matches snapshot 1`] = `[Function]`;
